accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/monotonic/entities/ContiguousIdBlock.java

```diff
@@ -73,8 +73,49 @@ public ContiguousIdBlock(String categoryId, String applicationInstanceId, long f
         this.lastCommitted = firstValue - 1;
     }
 
-    public ContiguousIdBlock nextBlock(String instanceId, long size, long nextBlockInterval) {
-        return new ContiguousIdBlock(categoryId, instanceId, lastValue + 1 + nextBlockInterval, size);
+    /**
+     * This method creates a new ContiguousIdBlock based on the previous block and interleaveInterval.
+     * In case of interleaveInterval == 0 the next block can start contiguously , but if it is not then
+     * the interleavingPoint is calculated by finding the range in which the lastvalue (previous block's endValue)
+     * falls.
+     *
+     * Below are the series of cases
+     *
+     * if earliestBlockStartValue = 0 , interleaveInterval = 50 , lastValue = 40 then,
+     * interleavingPoint = 50 and remainingSizeInRangeOfBlocks=9 so the nextBlockStartValue will be 41
+     * if earliestBlockStartValue = 0 , interleaveInterval = 50 , lastValue = 49 then,
+     * interleavingPoint = 50 and remainingSizeInRangeOfBlocks=0 so the nextBlockStartValue will be 100
+     *
+     * if earliestBlockStartValue = 0 , interleaveInterval = 50 , lastValue = 120 then,
+     * interleavingPoint = 150 and remainingSizeInRangeOfBlocks= 29 so the nextBlockStartValue will be 121
+     * if earliestBlockStartValue = 0 , interleaveInterval = 50 , lastValue = 130 and size is 25 then,
+     * interleavingPoint = 150 and remainingSizeInRangeOfBlocks= 19 so the nextBlockStartValue equals 131 and size = 19
+     *
+     * @param instanceId              - instance id for the machine
+     * @param size                    - size of the block to be reserved
+     * @param interleaveInterval      - specifies when to interleave and interval to interleave
+     * @param earliestBlockStartValue - earliest block start value refers to blockStartValue in properties file
+     * @return
+     */
+    public ContiguousIdBlock nextBlock(String instanceId, long size, long interleaveInterval,
+                                       long earliestBlockStartValue) {
+        long nextBlockStartValue = lastValue + 1;
+        if (interleaveInterval == 0) {
+            return new ContiguousIdBlock(categoryId, instanceId, nextBlockStartValue, size);
+        }
+        long interleavingPoint = ((lastValue - earliestBlockStartValue) / interleaveInterval + 1) * interleaveInterval;
```

Comment:
I think this line has a bug. The scenario with `blockStartValue != 0` is not tested with interleaveIntervals. Can you add a test next to testNextBlock?